iotics/lib/identity/api/advanced_api.py

```diff
@@ -713,18 +713,31 @@ def delegate_authentication(self, delegating_secrets: KeyPairSecrets, delegating
         subject_issuer, proof = AdvancedIdentityLocalApi.create_delegation_proof(delegating_issuer,
                                                                                  subject_doc,
                                                                                  subject_secrets)
+
+        existing_delegation = RegisterDocumentHelper.get_register_delegation_proof_by_controller(subject_issuer,
+                                                                                                 delegating_doc,
+                                                                                                 True)
+        if existing_delegation and (
+            not delegation_name or delegation_name == existing_delegation.name
+        ):
+            # Found an existing delegation with matching controller and name. Nothing to do.
+            return
+
+        if not delegation_name:
```

Comment:
`is not None`?

Reply:
N/A given [this](https://github.com/Iotic-Labs/iotics-identity-py/pull/15/files#r690395968)